### PR TITLE
Fix input checks of `TestMin` and `TestMax`

### PR DESCRIPTION
**Pull request by toslunar**
The checks have to be aware of `axis`:
```
y = x.max(axis=self.axis, keepdims=True)
if numpy.all((x > y - 2 * eps).sum(axis=self.axis) == 1):
    # ok
```

### CODE REVIEW

File: tests/chainer_tests/functions_tests/math_tests/test_minmax.py [hunk 1]
```diff
@@ -190,163 +108,80 @@ def test_duplicate_axis(self):
             functions.Max((0, 0))
 
     def test_pos_neg_duplicate_axis(self):
+        x_data = numpy.random.uniform(-1, 1, (3, 2, 4)).astype(numpy.float32)
+        x = chainer.Variable(x_data)
         with self.assertRaises(ValueError):
-            self.x.max(axis=(1, -2))
```
Comment (kmaehashi) [inline tests/chainer_tests/functions_tests/math_tests/test_minmax.py]: Good catch...

**Comment by kmaehashi** [comment]
LGTM!

**Comment by kmaehashi** [comment]
This is a fix F.min/max error of #3766

File: tests/chainer_tests/functions_tests/math_tests/test_minmax.py (hunk 1 above)
Comment (toslunar) [reply tests/chainer_tests/functions_tests/math_tests/test_minmax.py]: Thanks.  `ValueError` has been raised by `numpy.ndarray.max`, not by `chainer.functions.max`.

**Pull request merged**